### PR TITLE
Port to dune and upgrade opam metadata to 2.0

### PR DESCRIPTION
**Pull request by avsm**
Jbuilder is being deprecated soon, hence this upgrade.

### CODE REVIEW

File: dune-project [hunk 1]
```diff
@@ -0,0 +1,2 @@
+(lang dune 1.0)
```
Comment (edwintorok) [inline dune-project]: We've got dune 1.10 now, so this could use a higher number here, but this is fine too.

Comment (avsm) [reply dune-project]: The idea with the (lang) version in dune is to have the minimum dune version required for the language features.  Dune will prompt you with an error message and the updated version if you use something newer, so this helps keep the version constraint as flexible as possible.

**Comment by avsm** [comment]
Pushed a rebase with the DCO!

**Comment by avsm** [comment]
I'm not sure if xapi-project is still publishing to opam-repository, but a release of this would be handy if possible, since MirageOS depends on xenstore_transport in our dependency cone.

**Comment by lindig** [comment]
We are maintaining our own opam files but don't publish to the central opam-repository. Let me look into it.

**Comment by edwintorok** [comment]
I created a new tag https://github.com/xapi-project/ocaml-xenstore-clients/releases/tag/v1.1.0

**Comment by lindig** [comment]
We are currently fighting problems with Travis but the plan is to publish this to https://github.com/api-project/xs-opam and once it is there I would submit it to opam-repository.

**Pull request merged**